mdp/hinet/switchboard.py

```diff
@@ -688,3 +669,99 @@ def index_to_image(self, index):
+        self.field_dstr = field_dstr
+        self.in_channels_xy = in_channels_xy
+        self.field_channels_xy = field_channels_xy
+        self.out_channels = out_channels
+
+        out_channel_dim = (in_channel_dim * field_channels_xy[0] * field_channels_xy[1])
+        # check parameters for inconsistencies
+        for i, name in enumerate(["x", "y"]):
+            if field_channels_xy[i] > in_channels_xy[i]:
+                err = ("Number of field channels exceeds the number of "
+                       "input channels in %s-direction. "
+                       "This would lead to an empty connection list." % name)
+                raise RandomChannelSwitchboardException(err)
+
+        in_trans = CoordinateTranslator(*in_channels_xy)
+
+        # setup a look up table for accessing connected grid points.
+        # This incurs a one-time expensive computation when the node is created but
+        # keeps the execute call efficient computationally.
+        self._lut_conn = numx.zeros([self.in_channels_xy[1] - self.field_channels_xy[1] + 1,
+                                     self.in_channels_xy[0] - self.field_channels_xy[0] + 1,
+                                     out_channel_dim], dtype=numx.int32)
+
+        x_fields_origin = range(self.in_channels_xy[0] - self.field_channels_xy[0] + 1)
+        y_fields_origin = range(self.in_channels_xy[1] - self.field_channels_xy[1] + 1)
+
+        tot_conns = self._lut_conn.shape[0] * self._lut_conn.shape[1]
+        origins = numx.asarray(numx.meshgrid(x_fields_origin, y_fields_origin)).reshape(2, tot_conns).T
+
+        for i in xrange(tot_conns):
+            # inner loop over field
+            x_start_chan, y_start_chan = origins[i]
+            connections = numx.zeros([out_channel_dim], dtype=numx.int32)
+            first_out_con = 0
+            for y_in_chan in range(y_start_chan, y_start_chan + field_channels_xy[1]):
+                for x_in_chan in range(x_start_chan, x_start_chan + field_channels_xy[0]):
+                    first_in_con = (in_trans.image_to_index(x_in_chan, y_in_chan) * in_channel_dim)
+                    connections[first_out_con: first_out_con + in_channel_dim] = range(first_in_con,
+                                                                                       first_in_con + in_channel_dim)
+                    first_out_con += in_channel_dim
+            self._lut_conn[y_start_chan, x_start_chan] = connections
+
+        connections = self._new_connections()
+        super(RandomChannelSwitchboard, self).__init__(input_dim=(in_channel_dim * in_channels_xy[0] *
+                                                                  in_channels_xy[1]),
+                                                       connections=connections,
+                                                       out_channel_dim=out_channel_dim,
+                                                       in_channel_dim=in_channel_dim)
+
+    def _new_connections(self):
```

Comment:
I'd pick a more specific name, like "random_connections".
I'd either make this method purely functional (providing the required data as arguments) or fully go the object oriented route (by overwriting the connections in the method and calling it something like "update_random_connections").